src/invoice/signature.rs

```diff
@@ -69,7 +74,84 @@ impl Display for SignatureRole {
                 Self::Creator => "creator",
                 Self::Proxy => "proxy",
                 Self::Host => "host",
+                Self::Approver => "approver",
             }
         )
     }
 }
+
+/// A KeyRing contains a list of public keys.
+///
+/// The purpose of this keyring is to validate signatures. The keyring NEVER
+/// contains private keys.
+///
+/// The concepts are described in the signing-spec.md document for Bindle.
+#[derive(Serialize, Deserialize, Debug, Clone)]
+#[serde(rename_all = "camelCase")]
+pub struct KeyRing {
+    pub version: String,
+    pub key: Vec<KeyEntry>,
```

Comment:
nit: Should this be `keys`?

Reply:
Yeah, probably `keys`

Reply:
So... no... because TOML.

```toml
[[key]]
label = "Brigade pipeline at builder.example.com"
roles = ["proxy"]
key = "bb453q4..."
labelSignature = "dsaff678..."
```

Reply:
This is the pattern used throughout Bindle: Use singular names for entries, and then mirror that name in the code. I suppose we could debate changing this (and I did originally start with Serde aliases, and revert). But the current PR is consistent with the rest of Bindle.

Reply:
We _could_ rename the field on the struct, if `keys` made sense in code.

Reply:
Oh right, I remember that debate on parcels